internal/utils/test.go

```diff
@@ -105,3 +106,93 @@ func GetDataSourcesLength(state *terraform.State, tfVarName, dataSourceName stri
 
 	return instanceState, numAttribute, nil
 }
+
+func TestCheckResourceAttrExists(name, key string, isOptional bool) resource.TestCheckFunc {
```

Comment:
let's remove these functions since it doesn't offer too much
if an optional value is set or not set shouldn't be a huge test case for us
furthermore the code seems rather unmaintainable for us and i prefer not to have too many internal detail code in our repo
i do appreciate the level of effort though to get something like this working so kudos!

Reply:
sounds good!